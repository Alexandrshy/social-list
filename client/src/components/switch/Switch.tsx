import React, { useState, useEffect } from "react";

import { THEME } from "../../constants";
import useDarkMode from "../../hooks/use-dark-mode";

import s from "./Switch.module.css";

export const Switch: React.FC = () => {
  const [darkMode, prefersDarkMode] = useDarkMode();

  return (
    <div className={s.switch}>
      <label className={s.switch__label} htmlFor="switcher">
        <input
          className={s.switch__input}
          type="checkbox"
          id="switcher"
          checked={darkMode === THEME.DARK}
          onChange={() =>
            prefersDarkMode(darkMode === THEME.DARK ? THEME.LIGHT : THEME.DARK)
          }
        />
        <div className={s.switch__slider}>
          <div className={s.switch__circle} />
        </div>
      </label>
    </div>
  );
};
