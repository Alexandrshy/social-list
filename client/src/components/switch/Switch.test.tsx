import React from "react";
import renderer from "react-test-renderer";

import { Switch } from "./Switch";

beforeEach(() => {
  window.matchMedia = jest.fn();
});

describe("Switch component", () => {
  it("should render", () => {
    const tree = renderer.create(<Switch />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
