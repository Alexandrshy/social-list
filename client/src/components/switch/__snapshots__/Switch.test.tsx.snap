// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch component should render 1`] = `
<div
  className="switch"
>
  <label
    className="switch__label"
    htmlFor="switcher"
  >
    <input
      checked={false}
      className="switch__input"
      id="switcher"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="switch__slider"
    >
      <div
        className="switch__circle"
      />
    </div>
  </label>
</div>
`;
